nixos/modules/services/security/certmgr.nix

```diff
@@ -0,0 +1,200 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.certmgr;
+
+  specs = mapAttrsToList
+    (n: v:
+      let
+        name = n + ".json";
+        path = if isAttrs v then pkgs.writeText name (builtins.toJSON v) else v;
+      in
+        { inherit name; inherit path; })
+    cfg.specs;
+
+  allSpecs = pkgs.buildEnv {
+     name = "certmgr.d";
+     paths = optional (specs != []) (pkgs.linkFarm "certmgr-specs" specs);
+  };
```

Comment:
This seems to be the same as just `allSpecs = pkgs.linkFarm "certmgr.d" specs;`

Reply:
Of course. I just forgot to reduce it after all my testing. Fixed in https://github.com/NixOS/nixpkgs/pull/44556/commits/9082a6152a3bc2c27942908b529b05ab10efd224